### ISSUE TITLE
Make translator tests language-independent

### ISSUE THREAD

**Issue by zuphilip** [zotero/scaffold#21]
The tests can fail, if there is an English date saved and compared with a German date, e.g.
```
         -   "date": "June 1, 2014"
         +   "date": "Juni 1, 2014"
```
Is it possible to compare the standardized date format (i.e. 2014-06-01) instead?

**Comment by aurimasv** [comment]
I don't think we want to compare the interpreted date, but we can make sure that only en-US is used to generate and run tests by setting the locale to en-US before running/generating tests and reverting it right after. It's not a great option, since there's always a chance that we will fail to revert the locale under some condition, but given Scaffold's target audience, it's probably not as big of an issue.

**Comment by zuphilip** [comment]
Are you saying that "2014-06-01" is the interpreted date? I tought that would be the internal format of dates...

**Comment by aurimasv** [comment]
What you see in the test is what Zotero imports. It will then attempt to interpret the date and store it internally as YYYY-MM-DD

**Comment by aurimasv** [comment]
Actually it will store both formats. YYYY-MM-DD is not displayed to the user, just the raw imported date is shown.

**Comment by zuphilip** [comment]
Hm.. I think I am missing a step. For example this happens when importing RIS:
 * The RIS tag `Y1 - 1990/2/27` ...
 * ... will be shown in the test as `"issueDate": "February 27, 1990",`

Changing the locale preferences  for testing seems to be hacky and I would rather prefer not to do that. Anyway, if it is not possible to change the comparison in the test, I can also use my Firefox English Protable Version for testing, or ask you to do the tests;-)

**Comment by aurimasv** [comment]
RIS translator does pretty formatting for a date, which is locale dependent, so running the test and generating the test have to be done in the same locale. I'm not really sure which part of that is not clear.

**Comment by zuphilip** [comment]
Okay, and why is RIS translator doing that? For example another translator is saving the date as YYYY-MM-DD: https://github.com/zotero/translators/blob/master/semantics%20Visual%20Library.js#L241 (Didn't you advice me to do it like that?)

Maybe I am wrong, but wouldn't it be better to import the date as YYYY-MM-DD whenever possible and just render it in the client in the language the user choose?

**Comment by dstillman** [comment]
The date field can contain anything (e.g., "Summer 2014"), so it can't be limited to that, but we could say that if there's an ISO8601 date and nothing else, it gets rendered nicely.

**Comment by dstillman** [comment]
(And this would address a long-standing complaint about varying date formats in the middle pane.)

**Comment by aurimasv** [comment]
I like that, but it may be confusing for the user when editing the field. Though I suppose it's not terribly unusual to be editing editing data in one format and have it displayed in a different format (e.g. dates in excel)

>why is RIS translator doing that?

I'm not sure. That dates way way back. The formatting should really be done in Zotero UI (as Dan suggests) and the data should be stored in a structured format unless the user specifically puts something else in there. Imagine sharing an item with locale-dependent date formatting in a group. It wouldn't make sense to show a German date to someone using en-US locale.

**Comment by dstillman** [comment]
We could possibly use an HTML datepicker with a locale-aware free-form field.
